gradle-idea-configuration/src/main/java/com/palantir/gradle/ideaconfiguration/UpdateExternalDependenciesXml.java

```diff
@@ -0,0 +1,125 @@
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+public abstract class UpdateExternalDependenciesXml extends DefaultTask {
+    @Nested
+    public abstract SetProperty<PluginDependency> getDependencies();
+
+    @OutputFile
+    public abstract RegularFileProperty getOutputFile();
+
+    public UpdateExternalDependenciesXml() {
+        getOutputFile().set(getProject().file(".idea/externalDependencies.xml"));
+    }
+
+    @TaskAction
+    public final void updateXml() {
+        Set<PluginDependency> dependencies = getDependencies().get();
+        File outputFile = getOutputFile().get().getAsFile();
+
+        if (dependencies.isEmpty()) {
+            return;
+        }
+
+        XmlMapper xmlMapper = new XmlMapper();
+        xmlMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        List<PluginDependencyXml> existingPluginXmls = readExistingPluginXmls(xmlMapper, outputFile);
+        List<PluginDependencyXml> newPluginXmls = toXmlDependencies(dependencies);
+        List<PluginDependencyXml> merged = mergePluginXmls(existingPluginXmls, newPluginXmls);
+
+        writeMergedXml(xmlMapper, outputFile, merged);
+    }
+
+    private static List<PluginDependencyXml> readExistingPluginXmls(XmlMapper xmlMapper, File outputFile) {
+        if (!outputFile.exists()) {
+            return List.of();
+        }
+        try {
+            ProjectXml existingProject = xmlMapper.readValue(outputFile, ProjectXml.class);
+            if (existingProject != null && existingProject.component() != null) {
+                return existingProject.component().plugins();
+            }
+        } catch (IOException e) {
+            // Ignore and return empty
+        }
+        return List.of();
+    }
+
+    private static List<PluginDependencyXml> toXmlDependencies(Set<PluginDependency> deps) {
+        return deps.stream().map(PluginDependencyXml::from).collect(Collectors.toList());
+    }
+
+    private static List<PluginDependencyXml> mergePluginXmls(
+            List<PluginDependencyXml> existing, List<PluginDependencyXml> incoming) {
+        return java.util.stream.Stream.concat(existing.stream(), incoming.stream())
+                .collect(Collectors.toMap(
+                        PluginDependencyXml::id, dep -> dep, UpdateExternalDependenciesXml::pickHigherVersion))
+                .values()
+                .stream()
+                .sorted(Comparator.comparing(PluginDependencyXml::id))
+                .collect(Collectors.toList());
+    }
+
+    private static PluginDependencyXml pickHigherVersion(
+            PluginDependencyXml firstDependency, PluginDependencyXml secondDependency) {
+        String firstVersion = firstDependency.minVersion();
```

Comment:
maybe we can switch these to Optionals, otherwise we can use optionals here: eg. `Optional.ofNullable(firstDependency.minVersion())`